json-schema-converter/src/main/java/io/confluent/connect/json/JsonSchemaData.java

```diff
@@ -1138,29 +1138,32 @@ private Schema allOfToConnectSchema(
   static class SchemaWrapper extends SchemaBuilder {
 
     private final SchemaBuilder builder;
+    // Optional that overrides the one in builder
+    private boolean optional;
     // Parameters that override the ones in builder
     private final Map<String, String> parameters;
 
-    public SchemaWrapper(SchemaBuilder builder) {
+    public SchemaWrapper(SchemaBuilder builder, boolean optional) {
       super(Type.STRUCT);
       this.builder = builder;
+      this.optional = optional;
       this.parameters = new LinkedHashMap<>();
     }
 
     @Override
     public boolean isOptional() {
-      return builder.isOptional();
+      return optional;
     }
 
     @Override
     public SchemaBuilder optional() {
-      builder.optional();
+      optional = true;
       return this;
     }
 
     @Override
     public SchemaBuilder required() {
-      builder.required();
+      optional = false;
```

Comment:
Necessary? Some one can do optional().required()? Any test exercising it?

Reply:
`optional` and `required` just toggle the optionality.  See https://kafka.apache.org/28/javadoc/org/apache/kafka/connect/data/SchemaBuilder.html#required()